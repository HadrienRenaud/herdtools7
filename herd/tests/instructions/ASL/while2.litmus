ASL while2

{ }

func main() => integer
begin
  var x: integer;
  repeat
    while FALSE do
      let z = 2 ;
    end;
    x = 2;
  until x >= 2 ;

  return 0;
end;

forall (0:main.0.z=0 /\ 0:main.0.x=2)
