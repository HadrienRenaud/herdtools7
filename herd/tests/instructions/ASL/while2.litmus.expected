Test while2 Required
States 1
0:main.0.x=2; 0:main.0.z=0;
Ok
Witnesses
Positive: 1 Negative: 0
Condition forall (0:main.0.z=0 /\ 0:main.0.x=2)
Observation while2 Always 1 0
Hash=8a1dc35e9dc21b64f3d9dda4437921f4

